src/libmeasurement_kit/ooni/utils.cpp

```diff
@@ -162,5 +164,58 @@ std::string scrub(std::string s, std::string real_probe_ip) {
     return s;
 }
 
+void compare_headers_response(http::Headers headers,
+                             Var<http::Response> response, Var<report::Entry> entry,
+                             Var<Logger> logger) {
+    if (response->body.empty()) {
+        logger->warn("empty response body");
+        (*entry)["tampering"]["total"] = true;
+        (*entry)["tampering"]["request_line_capitalization"] = true;
+        return;
+    }
+
+    nlohmann::json resp;
+    try {
+        resp = nlohmann::json::parse(response->body);
+    } catch (const std::invalid_argument &) {
+        logger->warn("response body not valid JSON");
+        (*entry)["tampering"]["total"] = true;
+        (*entry)["tampering"]["request_line_capitalization"] = true;
+        return;
```

Comment:
Be careful to ensure that the keys `header_field_name` and `header_field_diff` are written also if you return early.